crates/iroha_wasm_builder/src/lib.rs

```diff
@@ -189,9 +201,9 @@ mod internal {
             })
         }
 
-        fn build_options() -> impl Iterator<Item = &'static str> {
+        fn build_options(&self) -> impl Iterator<Item = &'static str> {
             [
-                "--release",
+                if self.release { "--release" } else { "" },
```

Comment:
I think some CI workflows should keep `--release` build, at least `iroha2-release.yml` and maybe `iroha2-custom-image.yml` @BAStos525  ?

Reply:
In `iroha2-release.yml` we use the tag taken from `github.ref_name` - it's the tag name. Should any changes be introduced?

Reply:
Only `*.wasm` quality is discussed here, which would only affect e.g. [this line](https://github.com/hyperledger-iroha/iroha/blob/70e5aacf3290f4b143018bc6f21ce5dcecf38d18/.github/workflows/iroha2-release.yml#L22)

Reply:
CI will use `release` due to the need for additional resources (memory, fuel) and adjusted timeouts. Nevertheless, the debug flag can be set when building WASM.

Reply:
> CI will use `release` due to the need for additional resources (memory, fuel) and adjusted timeouts. Nevertheless, the debug flag can be set when building WASM.

I would like CI to use `test` when testing things and `release` (or `deploy`) when deploying. Yes, this will require you to adjust timeouts and limits in the `test_network` (don't modify parameters in `genesis.json`)